### PR TITLE
Release v2.0.0.b4

### PR DESCRIPTION
**Pull request by Gobot1234**
Release planning for 2.0.0b4

Things I think should be merged before the release:
- #280
- #281
- #304
- #187
- #293

TODO:
- [x] Make sure that the release date in the Changelog matches today's date before merging.
- [x] Relock the poetry.lock
- [ ] Publish the draft v2.0.0-beta4 release right after merging.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 
+## [2.0.0b4] - 202x-xx-xx
+
+### Breaking changes
+- The minimum Python version has been bumped to `3.6.2`
+
+- Fix compilation of fields with name identical to their type [#294](https://github.com/danielgtaylor/python-betterproto/pull/294)
+- Fix default values for enum service args [#299](https://github.com/danielgtaylor/python-betterproto/pull/299)
+- Always add `AsyncIterator` to imports if there are services [#264](https://github.com/danielgtaylor/python-betterproto/pull/264)
+- Allow parsing of messages from `ByteStrings` [#266](https://github.com/danielgtaylor/python-betterproto/pull/266)
```
Comment (kalzoo) [inline CHANGELOG.md:18]:
```suggestion

- **Breaking**: the minimum Python version has been bumped to `3.6.2`

- Always add `AsyncIterator` to imports if there are services [#264](https://github.com/danielgtaylor/python-betterproto/pull/264)
- Allow parsing of messages from `ByteStrings` [#266](https://github.com/danielgtaylor/python-betterproto/pull/266)

- Fix compilation of fields with name identical to their type [#294](https://github.com/danielgtaylor/python-betterproto/pull/294)
- Fix default values for enum service args [#299](https://github.com/danielgtaylor/python-betterproto/pull/299)
```

Comment (kalzoo) [reply CHANGELOG.md:18]: (Trying to match the style currently in use, although no recent changes are specifically called out as breaking)

**Review by kalzoo — CHANGES_REQUESTED** [review]

**Comment by kalzoo** [comment]
Hey @Gobot1234 I'd be happy to release how it is now (thanks for the 3.10 fix!) and then we can do a quick release afterwards once the others are in. Helps get `optional` out the door and then we're piling less into one release. Thoughts?

**Comment by jdlourenco** [comment]
@kalzoo releasing support for optional would be much appreciated! 👍

**Comment by Gobot1234** [comment]
@kalzoo sounds good to me

**Review by kalzoo — APPROVED** [review]
Thanks!

**Comment by lazytype** [comment]
Thanks everyone!

**Pull request merged**